src/core/utils/ping_sender.cpp

```diff
@@ -199,6 +199,7 @@ void PingSender::HandleIcmpReceive(const Message &          aMessage,
 {
     Reply    reply;
     uint32_t timestamp;
+    bool timerWasRunning;
```

Comment:
```suggestion

        VerifyOrExit(mTimer.IsRunning())
```

`PingSender` is not expecting any ping reply if `mTimer` is not running ?

Reply:
After timer stops, PingSender can still handle incoming echo replies and print its RTT, but it won't print statistics.

Reply:
> After timer stops, PingSender can still handle incoming echo replies and print its RTT, but it won't print statistics.

But these echo replies are unlikely for echo requests sent by `PingSender` since timeout has expired.

Reply:
Done